src/main.cpp

```diff
@@ -1229,8 +1232,8 @@ class ATTR_DLL_LOCAL SubtitleSampleReader : public SampleReader
 
   AP4_Sample m_sample;
   AP4_DataBuffer m_sampleData;
-  AP4_ByteStream* m_input{nullptr};
-  KodiAdaptiveStream* m_adaptiveStream{nullptr};
```

Comment:
Just a thought, doesn't have to be for this PR but thought I'd mention:
I know this is renamed to match the derived class, my question is do we really need this derived class at all? Yes, in Nexus now this derived class holds the rep chooser and also having download functions defined here made creating the test classes a little easier for me but I can't see why it was made this way?

Reply:
I have not check it in depth, but i agree that we could remove this derived class and merge it into the base class

Reply:
Cool. We'll bring this in for now as-is and look at that later